### PR TITLE
Feature/15872 map payload to check window

### PR DESCRIPTION
**Pull request by jon-shipley**
Pupil authentication changes, in the admin app:

* pupil auth requires a current check window
* the attempt meta data is stored in a new collection called `checks` which stores:
   - pupil id
   - checkwindow id
   - check form id
   - check start date
   - unique check code for this attempt

### CODE REVIEW

**Comment by jon-shipley** [comment]
Looking for 1 dev review

**Comment by GuyHarwood** [comment]
@activemq @sinhaamit1980 when writing the tests for this can you please assert the values entered into the database, as pupil login and mapping of check data to a specific check window is very important.

File: pupil-spa/test/features/step_definitions/pupil_auth_steps.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  current_time = Time.now
+  wait_until {page.evaluate_script('window.localStorage.getItem("pupil");')}
+  storage_pupil = JSON.parse page.evaluate_script('window.localStorage.getItem("pupil");')
+  stored_check = MongoDbHelper.get_pupil_check_metadata(storage_pupil['checkCode'])
+  expect(stored_check['updatedAt'].strftime('%d-%m-%y %H:%M')).to eql current_time.utc.strftime('%d-%m-%y %H:%M')
+  expect(stored_check['createdAt'].strftime('%d-%m-%y %H:%M')).to eql current_time.utc.strftime('%d-%m-%y %H:%M')
+  expect(stored_check['pupilId']).to eql @pupil_information['_id']
+  check_window_id = stored_check['checkWindowId'].to_s
+  MongoDbHelper.get_check_window(check_window_id)
+  expect(MongoDbHelper.get_check_window(check_window_id)['endDate'] > current_time).to be_truthy
```
Comment (jon-shipley) [inline pupil-spa/test/features/step_definitions/pupil_auth_steps.rb]: Do you want to check that the start date is in the past too?

File: pupil-spa/test/features/step_definitions/pupil_auth_steps.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  questions_recieved = JSON.parse page.evaluate_script('window.localStorage.getItem("questions");')
+  questions_expected = MongoDbHelper.get_form(100)['questions']
+  expect(questions_expected).to eql questions_recieved.map{|b| {'f1' => b['factor1'], 'f2' =>  b['factor2']}}
```
Comment (jon-shipley) [inline pupil-spa/test/features/step_definitions/pupil_auth_steps.rb]: v. tidy 👍

File: pupil-spa/test/features/step_definitions/pupil_auth_steps.rb (hunk 1 above)
Comment (activemq) [reply pupil-spa/test/features/step_definitions/pupil_auth_steps.rb]: Done!

**Pull request merged**